app/actions/battles.py

```diff
@@ -0,0 +1,8 @@
+def duel(knight1: dict, knight2: dict) -> dict:
+    knight1["hp"] -= (knight2["power"] - knight1["add_some_abilities"])
+    knight2["hp"] -= (knight1["power"] - knight2["add_some_abilities"])

```

Comment:
Similar to the previous line, ensure that 'knight1["power"]' is not less than 'knight2["add_some_abilities"]' to avoid increasing 'knight2["hp"]'.